hydrolib/core/dflowfm/polyfile/models.py

```diff
@@ -106,7 +129,53 @@ def _get_serializer(cls) -> Callable:
 
     @classmethod
     def _get_parser(cls) -> Callable:
-        # TODO Prevent circular dependency in Parser
+        # Prevent circular dependency in Parser
         from .parser import read_polyfile
 
         return read_polyfile
+
+    @property
+    def x(self) -> List[float]:
+        """X-coordinates of all points in the PolyFile."""
+        return [point.x for obj in self.objects for point in obj.points]
+
+    @property
+    def y(self) -> List[float]:
+        """Y-coordinates of all points in the PolyFile."""
+        return [point.y for obj in self.objects for point in obj.points]
+
+    def get_z_sources_sinks(self) -> Tuple[float, List[float]]:
+        """
+        Get the z values of the source and sink points from the polyline file.
+
+        Returns:
+            z_source, z_sinkA: Tuple[float, List[float]]:
+            If the polyline has data (more than 3 columns), then both the z_source and z_sink will be a list of two values.
+            Otherwise, the z_source and the z_sink will be a single value each.
+
+        Examples:
+        in case the polyline has 3 columns:
+            >>> polyline = PolyFile("tests/data/input/source-sink/leftsor.pliz")
+            >>> z_source, z_sink = polyline.get_z_sources_sinks()
+            >>> print(z_source, z_sink)
+            [-3] [-4.2]
+
+        in case the polyline has more than 3 columns:
+            >>> polyline = PolyFile("tests/data/input/source-sink/leftsor-5-columns.pliz") #Doctest: +SKIP
+            >>> z_source, z_sink = polyline.get_z_sources_sinks()
+            >>> print(z_source, z_sink)
+            [-3, -2.9] [-4.2, -5.35]
+        """
+        has_data = True if self.objects[0].points[0].data else False
+
+        z_source_sink = []
+        for elem in [0, -1]:
+            point = self.objects[0].points[elem]
+            if has_data:
```

Comment:
Also safeguard against `not have_z`

Reply:
there is a property `has_z_value` which can be used outside the method where the user can know if the polyline has a z value if yes the get_zsource_sink then can be used

Reply:
Now the `get_z_sources_sinks` will return a `[None], [None]` if the polyline does not have a z value as the `get_z_sources_sinks` just calls the `point.z` and for a polyline that does not have a z value the z value is assigned to None

Reply:
I have added documentation (Note in the `PolyFile.get_zsources_sinks`, also an examples) and a test `tests/dflowfm/poolyfole/test_polyline_models.py::TestGetZSourcesSinks.test_get_z_sources_sinks_no_z_values`